awx/main/tasks/facts.py

```diff
@@ -67,54 +73,73 @@ def start_fact_cache(hosts, destination, log_data, timeout=None, inventory_id=No
     msg='Inventory {inventory_id} host facts: updated {updated_ct}, cleared {cleared_ct}, unchanged {unmodified_ct}, took {delta:.3f} s',
     add_log_data=True,
 )
-def finish_fact_cache(hosts_cached, destination, facts_write_time, log_data, job_id=None, inventory_id=None):
+def finish_fact_cache(destination, log_data, job_id=None, inventory_id=None):
     log_data['inventory_id'] = inventory_id
     log_data['updated_ct'] = 0
     log_data['unmodified_ct'] = 0
     log_data['cleared_ct'] = 0
 
-    hosts_cached = sorted((h for h in hosts_cached if h.id is not None), key=lambda h: h.id)
+    summary_path = os.path.join(destination, 'host_cache_summary.json')
+    if not os.path.exists(summary_path):
+        system_tracking_logger.error(f'Missing summary file at {summary_path}')
```

Comment:
I looked at your test failures, and was awfully perplexed. It seems that what is making this very difficult to debut is the use of `system_tracking_logger` as opposed to `logger`. The former will present the logs in normal logs. I would avoid the former except for the very narrow cases where it is already being used. Because of this, we're not getting legitimate output from tests, which should show this log is being emitted, which is likely the current problem you're on.

Reply:
I believe this has been addressed!